src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -776,19 +776,24 @@ static int do_map(int argc, const char *argv[])
 
 static int do_unmap()
 {
+  
+  int r = 0;
+
   int nbd = open_device(devpath.c_str());
   if (nbd < 0) {
     cerr << "rbd-nbd: failed to open device: " << devpath << std::endl;
     return nbd;
   }
 
-  if (ioctl(nbd, NBD_DISCONNECT) < 0) {
-    cerr << "rbd-nbd: the device is not used" << std::endl;
+  r = ioctl(nbd, NBD_DISCONNECT);
+  if (r < 0) {
+      r = -errno;
+      cerr << "rbd-nbd: disconnect failed: " << cpp_strerror(-r) << std::endl;
```

Comment:
@trociny， I considered your suggestion， I also consider the implementation logic of nbd in the kernel， rbd-nbd: the device is not used  looks nicer than rbd-nbd: disconnect failed: (22) Invalid argument， I have adjusted the code，  thank you！